pallets/corporate-actions/src/distribution/mod.rs

```diff
@@ -349,12 +349,20 @@ decl_module! {
         #[weight = <T as Trait>::DistWeightInfo::reclaim()]
         pub fn reclaim(origin, ca_id: CAId) {
             // Ensure DID is the dist creator, they haven't reclaimed, and that expiry has passed.
-            let did = <Identity<T>>::ensure_perms(origin)?;
+            let did = <Identity<T>>::ensure_perms(origin.clone())?;
```

Comment:
This is already done in `ensure_agent_asset_perms ` so this line should be removed.